### ISSUE TITLE
Support for multiple Kafka topic subscription

### ISSUE THREAD

**Issue by samjdacanay** [smallrye/smallrye-reactive-messaging#424]
Right now, as it seems, the SmallRye Kafka Topic channels support a single topic per channel. In the [Kafka Source](https://github.com/smallrye/smallrye-reactive-messaging/blob/master/smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/impl/KafkaSource.java) it parses the Topic from a Configuration as a String.

Since the implementation is based on the Vert.x Kafka Consumer interface, per their documentation, it would be possible to support a [pattern for the Kafka Topic subscription](https://vertx.io/docs/vertx-kafka-client/java/#_receiving_messages_from_a_topic_joining_a_consumer_group).

I propose that if the topic config starts with a special character (like "~", similar to [Spotbugs](https://spotbugs.readthedocs.io/en/stable/filter.html)), we can assume that it is a Regular Expression and pass in a compiled Pattern to the subscribe method instead. 

Possible Impacts:
* Concurrency in the Kafka Source?
* Pattern compilation failure

**Comment by samjdacanay** [comment]
Actually now seeing that the Kafka Consumer is technically using the reactivex Kafka Consumer, not the one from the [Vert.x Kafka Client](https://vertx.io/docs/vertx-kafka-client/java/#_receiving_messages_from_a_topic_joining_a_consumer_group) that supports Patterns in the subscription. Trying to investigate if this is even currently possible...

**Comment by samjdacanay** [comment]
Upon closer inspection, it seems like these [subscribe methods](https://github.com/vert-x3/vertx-kafka-client/blob/master/src/main/java/io/vertx/kafka/client/consumer/KafkaConsumer.java) are actually ignored for the ReactiveX Code Generation. This seems to have been done in response to https://github.com/vert-x3/vertx-kafka-client/issues/88. 

We might have to update the vertx kafka client in order to be able to support Regex Subscription in here...

**Comment by cescoffier** [comment]
Fixed by #598.

**Comment by Wattos** [comment]
@cescoffier Thank you for updating the topic. I looked at the implementation and at the PR of #598 and do not fully understand how this issue is addressed.

I do not see any possibility to set a regex for a topic to connect to multiple kafka topics at the same time. I checked the documentation (https://github.com/smallrye/smallrye-reactive-messaging/blob/master/documentation/src/main/doc/modules/connectors/partials/META-INF/connector/smallrye-kafka-incoming.adoc) and I did not find any reference to be able to connect to multiple topics at the same time through a regex.

E.g. I would like to subscribe to the following topics:

`.*\.events`
which would subscribe to the following:

```
chats.events
users.events
groups.events
```
etc...
